### ISSUE TITLE
CORS: Missing Allow Origin

### ISSUE THREAD

**Issue by najtin** [thecodingmachine/workadventure#641]
What i did:
git clone ...
git checkout master
docker-compose up

I manually accepted the certificates for 
play.workadventure.localhost
api.workadventure.localhost
maps.workadventure.localhost
workadventure.localhost

But i still get this

![grafik](https://user-images.githubusercontent.com/42641080/105555382-8efd8400-5d09-11eb-8f52-ac0bc0a18b08.png)

any ideas on how to fix this? I am just getting into the project. So i am not really familiar with the code.

**Comment by Neph-Oo** [comment]
Your browser does not allow request on different ports. 
For test purposes, you can try installing an addon to allow CORS requests. Something like this one for example : https://addons.mozilla.org/fr/firefox/addon/access-control-allow-origin/


Another (cleaner) way to achieve this is to setup a proxy (nginx) who will redirect request to workadventure. You will need to edit docker-compose.yaml file to make workadventure listen on ports other than 80/443.

It's probably possible to do the same thing with traefik/docker but I don't know how.

**Comment by gmoirod** [comment]
There is a traefik in docker-compose to avoid this already. 
Thing is there is requests or redirects to play.workadventure.localhost:8080 instead of play.workadventure.localhost.
I think a fix need to be done in code

**Comment by cweiske** [comment]
See https://enable-cors.org/

**Comment by loelkes** [comment]
You can just allow CORS in Traefik with the following tags for the pusher container:
```
- "traefik.http.middlewares.cors-headers.headers.accesscontrolalloworigin=origin-list-or-null"
- "traefik.http.routers.pusher-ssl.middlewares=cors-headers@docker"
```

**Comment by Neph-Oo** [comment]
@gmoirod  is probably right too. 

I noticed that when I try to connect to a private URL (URL in the form play.foo.bar/@/[organization]/[world]/[map]), WA connects to port 8080 and fails if CORS is not allowed. 
It's not the case with the anonymous/public URL (URL in the form play.foo.bar/_/[instance]/[server]/[path to map]).
I don't know if it's expected (software in development mode) or not.

But anyway, does private URL work in the self hosted version of workadventure ?
I don't think that's the case.

**Comment by loelkes** [comment]
This has nothing to do with the development mode. There is a missing configuration (explained in #547): the server does not know it's behind a reverse proxy and the clients assume they must connect to :8080, the listening port of the server.

**Comment by gmoirod** [comment]
> But anyway, does private URL work in the self hosted version of workadventure ?
Was wondering the same and I would say no. Organization/World/Map management is done by the admin interface. This admin component is not given in this repo. This must be the proprietary part of the paid model.

**Comment by moufmouf** [comment]
Hey!

CORS is correctly configured in the docker-compose file.

Usually, CORS errors are due to underlying HTTP 500 errors (that fail to add the CORS header in the response). Your best bet is to have a look at the docker logs to see why there is a problem. Do not hesitate to put the docker logs output here.

@gmoirod is correct. The Organization/World/Map management and private urls in general are done by the admin interface which indeed is part of the paid model.

**Comment by vanhaakonnen** [comment]
Hi guys :)

I have the same problem... I tried so set up everything on one machine that is public available. 

When I connect with http everything is working except video and voice (that seems to require https.) With HTTPS I also get the CORS message. One message with port 443 and the other error messages with https and port 8080. 

I tried the fix with the three line of code in the docker-compose.yaml file for the pusher and front container - but it´s not working. currently I am not using any kind of proxy.

do do have an idea for me? :)
